social-network/src/main/java/com/levi9/socialnetwork/Service/impl/RegistrationServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+package com.levi9.socialnetwork.Service.impl;
+
+import com.levi9.socialnetwork.Exception.ResourceNotFoundException;
+import com.levi9.socialnetwork.Model.ConfirmationToken;
+import com.levi9.socialnetwork.Model.EStatus;
+import com.levi9.socialnetwork.Model.User;
+import com.levi9.socialnetwork.Security.authority.JWToken;
+import com.levi9.socialnetwork.Service.ConfirmationTokenService;
+import com.levi9.socialnetwork.Service.EmailService;
+import com.levi9.socialnetwork.Service.RegistrationService;
+import com.levi9.socialnetwork.Service.UserService;
+import com.levi9.socialnetwork.dto.RegistrationRequestDTO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.time.LocalDateTime;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Service
+public class RegistrationServiceImpl implements RegistrationService {
+
+    @Autowired
+    private JWToken jwToken;
+
+    @Autowired
+    private EmailService emailService;
+
+    @Autowired
+    private UserDetailsService userDetailsService;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    private ConfirmationTokenService confirmationTokenService;
+
+    @Autowired
+    private UserService userService;
+
+    public static final Pattern VALID_EMAIL_ADDRESS_REGEX =
+            Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$", Pattern.CASE_INSENSITIVE);
+
+    public static boolean validate(String emailStr) {
+        Matcher matcher = VALID_EMAIL_ADDRESS_REGEX.matcher(emailStr);
+        return matcher.find();
+    }
+
+    public static boolean validateRegistrationRequest(RegistrationRequestDTO registrationRequestDTO){
+        if (registrationRequestDTO.getName().isBlank() || registrationRequestDTO.getSurname().isBlank() ||
+                registrationRequestDTO.getPassword().length() < 6 || registrationRequestDTO.getRepeatedPassword().length() < 6 ||
```

Comment:
Extract 6 as constant PASSWORD_MIN_LENGTH or something similar. If a change of length is needed, you will change it only in one place.